From 1480a6555c67824d15474f2714ad68939c62ca9c Mon Sep 17 00:00:00 2001
From: Honore Tricot <honorex.tricot@intel.com>
Date: Mon, 17 Dec 2012 11:36:44 +0200
Subject: [PATCH 74/78] [PORT FROM JBMR0] wl12xx: disable elp sleep while in
 plt mode

BZ: 75310

DO NOT UPSTREAM
adapted from upstream patch (5cf0fe31abaff7a752cf72537702230c63b6116c)

We now disable elp sleep during plt mode to allow normal operation of
plt tools such as calibrator.

Having elp_sleep enabled during plt mode is actually not required and
in fact it distupt plt operations such as rx statistics etc...

Change-Id: Ibe4a3225b4588b2e13e9b296c9a9fd27549a50f8
Orig-Change-Id: I5e875914a51774e2286ae2148c312888bcb87ff6
Signed-off-by: Honore Tricot <honorex.tricot@intel.com>
Signed-off-by: Yair Shapira <yair.shapira@ti.com>
Reviewed-on: http://android.intel.com:8080/85617
Reviewed-by: Naoumenko, PaulX <paulx.naoumenko@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/ps.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/ps.c b/wl12xx-compat/drivers/net/wireless/wl12xx/ps.c
index 9d971eb..8d9c316 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/ps.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/ps.c
@@ -84,6 +84,10 @@ void wl1271_ps_elp_sleep(struct wl1271 *wl)
 	struct wl12xx_vif *wlvif;
 	u32 timeout;
 
+	/* We do not enter elp sleep in PLT mode */
+	if (wl->state == WL1271_STATE_PLT)
+		return;
+
 	/* we shouldn't get consecutive sleep requests */
 	if (WARN_ON(test_and_set_bit(WL1271_FLAG_ELP_REQUESTED, &wl->flags)))
 		return;
-- 
1.7.9

